### PR TITLE
Add ToDo app example

### PR DESCRIPTION
**Pull request by infosia**
READY FOR MERGE

- [TIMOB-19057](https://jira.appcelerator.org/browse/TIMOB-19057) Add ToDoList example
- [TIMOB-19208](https://jira.appcelerator.org/browse/TIMOB-19208) Fix: CommandBar disappears after closing overlapping Window
- [TIMOB-19213](https://jira.appcelerator.org/browse/TIMOB-19213) fireEvent fires only one listener

Will push new PR on [appcelerator-developer-relations/Sample.Todo](https://github.com/appcelerator-developer-relations/Sample.Todo) once this become stable.

~~Known issue: Switching between Tab doesn't update ListView data. On iOS ListView data is updated to the latest when you switch Tab. [TIMOB-19207](https://jira.appcelerator.org/browse/TIMOB-19207)~~

To test ToDo app: generate `ToDoList` project and then run `ToDoList.sln`.

```
$ titanium_mobile_windows/Tools/Scripts/generate_vs_project.sh ToDoList
```

### CODE REVIEW

**Comment by infosia** [comment]
windows_8_1

**Comment by infosia** [comment]
PR for Sample.Todo: https://github.com/appcelerator-developer-relations/Sample.Todo/pull/17

**Comment by dawsontoth** [comment]
looks good

File: Source/TitaniumKit/src/UI/Tab.cpp [hunk 1]
```diff
@@ -37,17 +38,32 @@ namespace Titanium
...
+		{
+			Titanium::UI::View::blur();
+
+			// Update Ti.UI.currentTab
+			const auto Ti = static_cast<JSObject>(get_context().get_global_object().GetProperty("Titanium"));
+			const auto UI = static_cast<JSObject>(Ti.GetProperty("UI")).GetPrivate<Titanium::UIModule>();
+			UI->set_currentTab(nullptr);
 		}
 
 		void Tab::open(const std::shared_ptr<OpenWindowParams>& options) TITANIUM_NOEXCEPT
```
Comment (sgtcoolguy) [inline Source/TitaniumKit/src/UI/Tab.cpp]: The docs state we take in a window arg and an options arg. I just see options here and it doesn't look like they're actually used...

**Pull request merged**